rootchain/samuel/samuel.go

```diff
@@ -0,0 +1,580 @@
+package samuel
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+
+	"github.com/0xPolygon/polygon-edge/crypto"
+	"github.com/0xPolygon/polygon-edge/e2e/framework"
+	"github.com/0xPolygon/polygon-edge/rootchain"
+	"github.com/0xPolygon/polygon-edge/rootchain/payload"
+	"github.com/0xPolygon/polygon-edge/rootchain/proto"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/umbracle/ethgo/abi"
+	googleProto "google.golang.org/protobuf/proto"
+)
+
+var (
+	errUnknownPayloadType = errors.New("unknown payload type")
+)
+
+// eventTracker defines the event tracker interface for SAMUEL
+type eventTracker interface {
+	// Start starts the event tracker from the specified block number
+	Start(uint64) error
+
+	// Stop stops the event tracker
+	Stop() error
+
+	// Subscribe creates a rootchain event subscription
+	Subscribe() <-chan rootchain.Event
+}
+
+// samp defines the SAMP interface for SAMUEL
```

Comment:
To be honest, I don't see what samp means. Please write what it stands for in comment when you use abbreviation.

Reply:
Explained 👍 

[1bb2688](https://github.com/0xPolygon/polygon-edge/pull/717/commits/1bb26880acfe111c2e85d889d152c8973e3b3367)